### PR TITLE
Sanitise data object before sending in defineRoute

### PR DESCRIPTION
**Pull request by laxa88**
Updates https://github.com/univapaycast/univapay-finance/issues/1244

Not really related to above ticket, but I found this issue while working on that ticket. 🙇

### CODE REVIEW

File: src/resources/Resource.ts [hunk 1]
```diff
@@ -76,11 +76,20 @@ export abstract class Resource extends EventEmitter {
...
             pathParams: Record<string, string> = {}
         ): Promise<B> {
+            /**
+             * Sanitises and ensures that the data is recreated as an object with default prototypes.
+             * In rare cases if the data was created with `Object.create(null)` (no prototypes),
+             * this will cause the data to be sent as `WebKitFormBoundary` instead of regular JSON object.
+             *
+             * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create#custom_and_null_objects
+             */
+            const data = { ...originalData };
```
Comment (phieronymus) [inline src/resources/Resource.ts:91]: You may conserve the `null` or `undefined` values for the data. And, in a general manner, the type of the data if not an object.

Comment (laxa88) [reply src/resources/Resource.ts:91]: oh right... 🤔 i didn't think about that. thanks! let me update this.

Comment (laxa88) [reply src/resources/Resource.ts]: added explicit object type check, hopefully this is more specific 🙇

**Review by phieronymus — APPROVED** [review]
Thanks. Merging this.

**Pull request merged**